test/create-request-client.js

```diff
@@ -373,102 +373,122 @@ describe('request client', () => {
     });
   });
 
-  it('should not omit empty params by default', () => {
-    const request = createAppRequestClient(input);
-    return request({
-      url: 'http://zapier-httpbin.herokuapp.com/get',
-      params: {
-        something: '',
-        cool: 'true'
-      }
-    }).then(responseBefore => {
-      const response = JSON.parse(JSON.stringify(responseBefore));
+  describe('adds query params', () => {
+    it('should not omit empty params by default', () => {
+      const request = createAppRequestClient(input);
+      return request({
+        url: 'http://zapier-httpbin.herokuapp.com/get',
+        params: {
+          something: '',
+          really: '{{bundle.inputData.really}}',
+          cool: 'true'
+        }
+      }).then(responseBefore => {
+        const response = JSON.parse(JSON.stringify(responseBefore));
 
-      response.json.args.something.should.eql('');
-      response.json.args.cool.should.eql('true');
-      response.status.should.eql(200);
+        response.json.args.something.should.eql('');
+        response.json.args.really.should.eql('{{bundle.inputData.really}}');
+        response.json.args.cool.should.eql('true');
+        response.status.should.eql(200);
 
-      const body = JSON.parse(response.content);
-      body.url.should.eql(
-        'http://zapier-httpbin.herokuapp.com/get?something=&cool=true'
-      );
+        const body = JSON.parse(response.content);
+        body.url.should.eql(
+          'http://zapier-httpbin.herokuapp.com/get?something=&really={{bundle.inputData.really}}&cool=true'
```

Comment:
It feels like a bug that we send `{{...}}` to a server. I don't see any possibility where we want to keep `{{...}}` in a request payload. Maybe it's reasonable to change the default behavior so when the `var` in `{{var}}` is undefined or null, we replace with an empty string? So here it would be `'http://zapier-httpbin.herokuapp.com/get?something=&really=&cool=true'. This is going to be a bug fix and a breaking change at the same time.

Reply:
@eliangcs I'll save this for the breaking change. For now, we'll keep the current behavior.

Reply:
@stevelikesmusic feel free to save this for a separate PR.

Good point on missing values in the request body. I'd expect different behaviors based on the request body format. If it's a JSON body, `null` should still be `null` and should never serialized as `''`. But if it's a form body, the situation is kind of like the query parameters - we're forced to serialize all the data types to strings, in which case, `null` should become `''`. Does that make sense?